File: native-android/app/src/main/java/com/iganapolsky/randomtimer/crash/CrashReportingService.kt
package com.iganapolsky.randomtimer.crash

import com.google.firebase.crashlytics.FirebaseCrashlytics
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class CrashReportingService @Inject constructor() {

    private val crashlytics: FirebaseCrashlytics by lazy {
        FirebaseCrashlytics.getInstance()
    }

    fun initialize() {
        // Crashlytics is automatically initialized via google-services.json
        // Enable collection (can be disabled for debug builds or user opt-out)
        crashlytics.isCrashlyticsCollectionEnabled = true
    }

    fun setUserId(userId: String) {
        crashlytics.setUserId(userId)
    }

    fun log(message: String) {
        crashlytics.log(message)
    }

    fun setCustomKey(key: String, value: String) {
        crashlytics.setCustomKey(key, value)
    }

    fun setCustomKey(key: String, value: Boolean) {
        crashlytics.setCustomKey(key, value)
    }

    fun setCustomKey(key: String, value: Int) {
        crashlytics.setCustomKey(key, value)
    }

    fun recordException(throwable: Throwable) {
        crashlytics.recordException(throwable)
    }

    fun sendUnsentReports() {
        crashlytics.sendUnsentReports()
    }
}

File: native-ios/RandomTimer/Sources/Services/CrashReportingService.swift
import Foundation
import FirebaseCrashlytics

/// Crash reporting service using Firebase Crashlytics
final class CrashReportingService: @unchecked Sendable {
    static let shared = CrashReportingService()

    private init() {}

    func initialize() {
        // Crashlytics is automatically initialized via GoogleService-Info.plist
        // Enable collection
        Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(true)
    }

    func setUserId(_ userId: String) {
        Crashlytics.crashlytics().setUserID(userId)
    }

    func log(_ message: String) {
        Crashlytics.crashlytics().log(message)
    }

    func setCustomValue(_ value: Any?, forKey key: String) {
        Crashlytics.crashlytics().setCustomValue(value, forKey: key)
    }

    func record(error: Error) {
        Crashlytics.crashlytics().record(error: error)
    }

    func sendUnsentReports() {
        Crashlytics.crashlytics().sendUnsentReports()
    }
}
